### PR TITLE
ui: crash when username is null

### PR DESCRIPTION
**Pull request by TLGINO**
* closes https://github.com/inveniosoftware/invenio-app-rdm/issues/2229

### CODE REVIEW

File: invenio_users_resources/entity_resolvers.py [hunk 1]
```diff
@@ -63,17 +63,27 @@ def pick_resolved_fields(self, identity, resolved_dict):
         avatar = current_users_service.links_item_tpl.expand(identity, fake_user_obj)[
             "avatar"
         ]
-        return {
+        obj = {
+            "identity": identity,
             "id": resolved_dict["id"],
-            "username": resolved_dict["username"],
+            "username": resolved_dict.get("username", ""),
```
Comment (ppanero) [inline invenio_users_resources/entity_resolvers.py:69]: @TLGINO Could you add a screenshot of how would the request be displayed now? What would appear in the `username` place? a placeholder? would it be empty? If the later, what do you think about defaulting to a mandatory field? (e.g. email, or full_name?)

File: invenio_users_resources/services/schemas.py [hunk 2]
```diff
@@ -67,10 +67,16 @@ class UserSchema(BaseRecordSchema, FieldPermissionsMixin):
     def is_self(self, obj):
         """Determine if identity is the current identity."""
         current_identity = self.context["identity"]
+
+        try:
+            _id = obj.id
+        except AttributeError:
+            _id = obj.get("id")
```
Comment (anikachurilova) [inline invenio_users_resources/services/schemas.py:74]: Question: can we do something like `obj.get("id") or obj.id` here?

**Review by anikachurilova — CHANGES_REQUESTED** [review]

File: invenio_users_resources/services/schemas.py (hunk 2 above)
Comment (TLGINO) [reply invenio_users_resources/services/schemas.py]: Done! much cleaner that way thx!

File: invenio_users_resources/entity_resolvers.py [hunk 3]
```diff
@@ -63,17 +63,27 @@ def pick_resolved_fields(self, identity, resolved_dict):
...
+            "preferences": resolved_dict.get("preferences", {}),
         }
 
+        resolved_dict["identity"] = identity
+
+        serialized_user = UserSchema(context=resolved_dict).dump(obj)
+
+        # Cannot serialize this field in the schema as it not part of the user schema
+        serialized_user["is_ghost"] = resolved_dict.get("is_ghost", False)
+        serialized_user["links"] = ({"avatar": avatar},)
```
Comment (anikachurilova) [inline invenio_users_resources/entity_resolvers.py:84]: `links` is part of `BaseRecordSchema` that `UserSchema` inherits from. Doesn't it dump this value when dumping the user?

Comment (TLGINO) [reply invenio_users_resources/entity_resolvers.py:84]: I got the impression it didn't dump the avatar field
le me re-test it to be sure

Comment (TLGINO) [reply invenio_users_resources/entity_resolvers.py:84]: nevermind it does get serialized correctly :/
willl push in a sec

**Review by anikachurilova — APPROVED** [review]

**Review by anikachurilova — APPROVED** [review]
LGTM!

**Review by slint — COMMENTED** [review]
One major point the serialization/permissions. This needs also thorough testing, since we're touching on very core parts of displaying users across the entire platform.

File: invenio_users_resources/entity_resolvers.py [hunk 4]
```diff
@@ -63,17 +63,27 @@ def pick_resolved_fields(self, identity, resolved_dict):
...
                 "full_name": profile.get("full_name", ""),
                 "affiliations": profile.get("affiliations", ""),
             },
+            "preferences": resolved_dict.get("preferences", {}),
             "links": {"avatar": avatar},
         }
 
+        resolved_dict["identity"] = identity
+
+        serialized_user = UserSchema(context=resolved_dict).dump(obj)
```
[the remark this answers is not in the document]
Comment (kpsherva) [reply invenio_users_resources/entity_resolvers.py:81]: we talked with @slint and we agreed to go ahead with the current state as long as the "unknown user" will be replaced with "anonymous user" in the ui, so do not address the comment above, we will ticketize it

**Pull request merged**